Automattic/pocket-casts-android
# Refactor bookmark change title deep linking

## Description

> [!note]
> This will be a series of PRs that move deep linking to a separate module that can be tested. The goal is to address #2405 but I don't want to do it blindly without having tests in order to not introduce regressions to deep linking.

This PR migrates changing bookmark title deep linking to the new module.

## Testing Instructions

You can test it by assigning `Add bookmark` action to your headphones, and then triggering it while the app is in background and choosing `Change title` option. If you can't test it this way do the following:

1. Create a bookmark.
2. Using app manger find an UUID of a bookmark.
3. Execute `adb shell am start -n au.com.shiftyjelly.pocketcasts.debug/au.com.shiftyjelly.pocketcasts.ui.MainActivity  -a INTENT_OPEN_APP_CHANGE_BOOKMARK_TITLE -e bookmark_uuid <BOOKMARK_UUID>`.
4. The bookmark change title page should show.

## Checklist
